Sming/Services/Yeelight/YeelightBulb.cpp

```diff
@@ -157,33 +156,29 @@ bool YeelightBulb::onResponse(TcpClient& client, char* data, int size)
 			p2 = source.length();
 		String buf = source.substring(p, p2);
 		p = unsigned(p2) + 2;
-		DynamicJsonBuffer jsonBuffer;
-		JsonObject& root = jsonBuffer.parseObject(buf);
-		bool parsed = root.success();
-		if (parsed)
+		DynamicJsonDocument doc(1024);
+		auto error = deserializeJson(doc, buf);
+		if (!error)
 		{
-			if (root.containsKey("id") && root.containsKey("result"))
+			long id = doc["id"] | -1;
+			if (id == propsId)
 			{
-				long id = root["id"];
-				if (id == propsId)
-				{
-					auto &result = root["result"].asArray();
-					String resp = result[0].asString();
+				JsonArray result = doc["result"];
+				String resp = result[0] | "";
+				if(resp.length()) {
 					parsePower(resp);
 				}
 			}
-			if (root.containsKey("method") && root.containsKey("params"))
+
+			String method = doc["method"] | "";
+			debugf("LED method %s received", method.c_str());
+			if (method == "props")
 			{
-				String method = root["method"].asString();
-				debugf("LED method %s received", method.c_str());
-				if (method == "props")
+				JsonObject result = doc["params"];
+				for (JsonObject::iterator it=result.begin(); it!=result.end(); ++it)
 				{
-					auto &result = root["params"].asObject();
-					for (JsonObject::iterator it=result.begin(); it!=result.end(); ++it)
-					{
-						if (strcmp(it->key, "power") == 0)
-							parsePower(it->value);
-					}
+					if (strcmp(it->key().c_str(), "power") == 0)
+						parsePower(it->value());
```

Comment:
I wonder why we're bothering with the iterator here, as we 're only after one value...